EventListener/RateLimitAnnotationListener.php

```diff
@@ -68,7 +69,13 @@ public function onKernelController(FilterControllerEvent $event)
 
         // Another treatment before applying RateLimit ?
         $checkedRateLimitEvent = new CheckedRateLimitEvent($event->getRequest(), $rateLimit);
-        $this->eventDispatcher->dispatch(RateLimitEvents::CHECKED_RATE_LIMIT, $checkedRateLimitEvent);
+        if(Kernel::VERSION_ID >= 40300) {
+            // Support for symfony listeners which are using the old event name.
+            $this->eventDispatcher->dispatch($checkedRateLimitEvent, RateLimitEvents::CHECKED_RATE_LIMIT);
+            $this->eventDispatcher->dispatch($checkedRateLimitEvent);
```

Comment:
Why is it dispatched twice?

Reply:
Since symfony 4.3 did not drop the support for the "traditional" event system, I guessed this was a way to have them both working in parallel.

(did not really check the "roadmap" on how to keep both systems running in symfony 4.3))

Reply:
It seems to me that  the first one should suffice: https://github.com/symfony/event-dispatcher/blob/master/EventDispatcher.php#L55. Symfony is using the event name you're passing, or the event class name if none is provided. Also check https://www.symfony-news.com/news/details/new-in-symfony-4-3-simpler-event-dispatching.